### Investigate null values in item_type_code_name

In [1]:
%run '../lib/libraries.ipynb'

dython              0.6.1


You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.


tinydb              4.2.0


You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.


In [2]:
filterDf = load_df('filterDf')

###### Create a dataframe of null observations

In [3]:
nullDf = filterDf.filter(F.col('item_type_code_name').isNull())

nullDf.count()

x observations

###### Number of null values

In [4]:
nullDf.select().count()

3264

- 3264
- close to the 3298 nulls in response correctness


###### Display any distinct values

In [5]:
distinct_val(nullDf)

0
assignment_max_attempts                1
assignment_attempt_number              1
is_manual_scoring_required         False
item_is_offline_scored             False
scoring_type_code           [unassigned]
assigned_item_status            assigned

- Looks like it wasn't answered
  - response_correctness = Null (None)
  - is a subset of the 3298 nulls in response correctness

- Will be resolved when the 3298 null in response_correctness is resolved

In [6]:
finish_todo('Investigate null values in item_type_code_name')